src/paths/distances.c

```diff
@@ -212,3 +212,34 @@ int igraph_radius(const igraph_t *graph, igraph_real_t *radius,
 
     return 0;
 }
+
+long int min(long int a, long int b) 
+{
+    return (a > b) ? b : a;
+}
+
+int igraph_center(const igraph_t *graph, igraph_vector_t *res,
+                  igraph_neimode_t mode) {
+
+    igraph_vector_t ecc;
+    igraph_vit_t vit;
+    igraph_vs_t vids = igraph_vss_all();
+
+    igraph_vector_init(&ecc, 0);
+
+    igraph_eccentricity(graph, &ecc, vids, mode);
+    IGRAPH_CHECK(igraph_vit_create(graph, vids, &vit));
+    if (!IGRAPH_VIT_END(vit)) {
```

Comment:
This loop is unnecessary. We only need to find the vertices with maximum eccentricity, so we only need to loop over the eccentricity vector.

Reply:
The idea of adding the condition `if (!IGRAPH_VIT_END(vit))` is to check for Null graph. If this is not correct, is there another way of doing it?

Reply:
Good point that we must check for the null graph, otherwise `vector_min` fails. Do this by checking the vertex count.